Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -508,13 +510,20 @@ private SessionInfo GetSessionInfo(
             ArgumentException.ThrowIfNullOrEmpty(deviceId);
 
             var key = GetSessionKey(appName, deviceId);
+            SessionInfo sessionInfo;
 
-            CheckDisposed();
-
-            if (!_activeConnections.TryGetValue(key, out var sessionInfo))
+            try
+            {
+                _activeConnectionsLock.EnterWriteLock();
```

Comment:
There is no benefit of using a ReaderWriterLock when you dont also lock read operations and only use it to lock write ones